offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -743,6 +815,75 @@ export class CkbHandler {
     }
   }
 
+  async parseUnlockTx(tx: Transaction): Promise<NervosUnlockAssetTxMetaData | null> {
+    if (tx.inputs.length < 2) {
+      return null;
+    }
+    const sudtTypescript = ForceBridgeCore.config.ckb.deps.sudtType.script;
+    const previousOutput = nonNullable(tx.inputs[0].previousOutput);
+    const preHash = previousOutput.txHash;
+    const txPrevious = await this.ckb.rpc.getTransaction(preHash);
+    if (txPrevious == null) {
+      return null;
+    }
+    const senderLockscript = txPrevious.transaction.outputs[Number(previousOutput.index)].lock;
+    const committeeMultisigLockscript = parseAddress(getOmniLockMultisigAddress());
+    const collectorAddress = getFromAddr();
+    const collectorLockscript = parseAddress(collectorAddress);
+    if (
+      !senderLockscript ||
+      senderLockscript.codeHash !== committeeMultisigLockscript.code_hash ||
+      senderLockscript.hashType !== committeeMultisigLockscript.hash_type ||
+      senderLockscript.args !== committeeMultisigLockscript.args
+    ) {
+      logger.error(`sender must be committee multisig cell`);
+      return null;
+    }
+
+    for (let i = 1; i < tx.inputs.length; i++) {
+      const thisPreviousOutput = nonNullable(tx.inputs[i].previousOutput);
+      const thisPreHash = thisPreviousOutput.txHash;
+      const thisTxPrevious = await this.ckb.rpc.getTransaction(thisPreHash);
+      if (thisTxPrevious == null) {
+        return null;
+      }
+      const thisSenderLockscript = thisTxPrevious.transaction.outputs[Number(thisPreviousOutput.index)].lock;
+      if (
+        !thisSenderLockscript ||
+        !(
+          (thisSenderLockscript.codeHash !== committeeMultisigLockscript.code_hash &&
+            thisSenderLockscript.hashType !== committeeMultisigLockscript.hash_type &&
+            thisSenderLockscript.args !== committeeMultisigLockscript.args) ||
+          (thisSenderLockscript.codeHash !== collectorLockscript.code_hash &&
+            thisSenderLockscript.hashType !== collectorLockscript.hash_type &&
+            thisSenderLockscript.args !== collectorLockscript.args)
+        )
+      ) {
+        logger.error(`inputs contain must be committee multisig or collector cell`);
+        return null;
+      }
+    }
+    const unlockMemoWitness = tx.witnesses[tx.witnesses.length - 1];
+    const unlockMemo = new UnlockMemo(new Reader(unlockMemoWitness).toArrayBuffer());
+    let chain;
+    let burnIds;
+    try {
+      chain = unlockMemo.getXchain();
+      const burnIdBytes: BytesVec = unlockMemo.getBurnIds();
+      burnIds = lodash.range(burnIdBytes.length()).map((i) => new Reader(burnIdBytes.indexAt(i).raw()).serializeJson());
+    } catch (e) {
+      logger.warn(`parse recipient unlock memo data error: ${e.message} ${e.stack}`);
+      return null;
+    }
+
+    const ethereumBurns: EthereumBurn[] = await this.db.getEthereumBurnByUniqueIds(burnIds);
```

Comment:
Don't get info from db, get it from tx.

Reply:
fixed, parse unlock info from transaction